sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -377,10 +377,61 @@ export interface EventData {
   };
 }
 
+/**
+ * Asserts that the provided data conforms to the `EventData` interface.
+ *
+ * This function performs runtime checks on the `data` object to ensure it matches the expected
+ * structure and types defined in the `EventData` interface. If any of the checks fail, it throws
+ * an error with a descriptive message indicating the mismatch.
+ *
+ * @param data - The data object to validate as `EventData`.
+ * @throws \{Error\} Throws an error if the data does not conform to the `EventData` interface.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertIsEventData(data: any): asserts data is EventData {
```

Comment:
nit: should we return true for valid case?

Reply:
No, it is a type of assertion function that errors when the test fails and void otherwise.

Reply:
Then the `: asserts data is EventData` part is not needed at all, right?  I thought the type guard may be useful in the future.

Reply:
[assertion function](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#assertion-functions) with the `asserts` keyword in the return type doesn't return. I think there is a confusion here between it and the type guard without the `asserts` keyword.

Reply:
For example, see [here](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgBrIN7IB4C5kDOYUoA5sgL4CwAULTAK4gJjAD2IywBqAFHsjggAngEp82LgTSZayecigQwDKJ0kBCALxbkTACYQYoCPuQAyc8jDCADhDYwcAOkk7dAIiIkQpDwG5aajoaRmZWDkECAmgwPgEhMXw4aNjpSW4ZDDkFYCdeDX5kbV0DIxMzS2s7BydsV2R3ZC9iMg9RUVkaBR7rAAsoNgB3ZBAIEYBRKEGoXg8QNjA0dsDuhWCKIA)

Reply:
You are right! I missed the `asserts`.